sdk/src/sdk/vdb/grpc/kuksa_val_v2/BrokerAsyncGrpcFacade.cpp

```diff
@@ -0,0 +1,119 @@
+/**
+ * Copyright (c) 2022-2024 Contributors to the Eclipse Foundation
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Apache License, Version 2.0 which is available at
+ * https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "BrokerAsyncGrpcFacade.h"
+
+#include "sdk/Logger.h"
+#include "sdk/grpc/GrpcCall.h"
+
+#include <grpcpp/channel.h>
+
+namespace velocitas::kuksa_val_v2 {
+
+BrokerAsyncGrpcFacade::BrokerAsyncGrpcFacade(std::shared_ptr<grpc::Channel> channel)
+    : m_stub{kuksa::val::v2::VAL::NewStub(channel)} {}
+
+void BrokerAsyncGrpcFacade::GetValues(
+    kuksa::val::v2::GetValuesRequest&&                                  request,
+    std::function<void(const kuksa::val::v2::GetValuesResponse& reply)> replyHandler,
+    std::function<void(const grpc::Status& status)>                     errorHandler) {
+    auto callData = std::make_shared<GrpcSingleResponseCall<kuksa::val::v2::GetValuesRequest,
+                                                            kuksa::val::v2::GetValuesResponse>>();
+
+    callData->m_request = std::move(request);
+
+    applyContextModifier(*callData);
+
+    auto grpcResultHandler = [callData, replyHandler, errorHandler](grpc::Status status) {
```

Comment:
`const auto grpcResultHandler` maybe better since `grpcResultHandler` should not be modified or reassigned?